subsys/llext/llext_load.c

```diff
@@ -261,14 +289,26 @@ static int llext_map_sections(struct llext_loader *ldr, struct llext *ext)
 				return -ENOEXEC;
 			}
 
-			if (mem_idx == LLEXT_MEM_BSS) {
+			/* Check if this region type is extendable */
+			switch (mem_idx) {
+			case LLEXT_MEM_BSS:
 				/* SHT_NOBITS sections cannot be merged properly:
 				 * as they use no space in the file, the logic
 				 * below does not work; they must be treated as
 				 * independent entities.
 				 */
 				LOG_ERR("Multiple SHT_NOBITS sections are not supported");
 				return -ENOTSUP;
+			case LLEXT_MEM_PREINIT:
+			case LLEXT_MEM_INIT:
+			case LLEXT_MEM_FINI:
+				/* These regions are not extendable and must be
+				 * referenced at most once in the ELF file.
+				 */
+				LOG_ERR("Region %d redefined", mem_idx);
+				return -ENOEXEC;
```

Comment:
wouldn't this be caught in line 211?

Reply:
No, line 211 checks that the _ELF section_ (`ldr->sect_map[shndx]`) was assigned a region before entering `llext_map_sections` - I added that log so the whole function prints a full ELF section table, including reserved sections found by `llext_find_tables`. 
This is instead checking if the _region_ has been already defined (ie. `ldr->sects[mem_idx]` - the array that really should be renamed `ldr->regions`...).